pkgs/stdenv/generic/setup.sh

```diff
@@ -341,27 +351,24 @@ declare -a allPlatOffsets=(-1 0 1)
 # implements.
 findInputs() {
     local -r pkg="$1"
-    local -ri hostOffset="$2"
-    local -ri targetOffset="$3"
+    local -r hostOffset="$2"
+    local -r targetOffset="$3"
 
     # Sanity check
-    (( "$hostOffset" <= "$targetOffset" )) || exit -1
+    (( "$hostOffset" <= "$targetOffset" )) || exit 1
 
-    local varVar="${pkgAccumVarVars[$hostOffset + 1]}"
-    local varRef="$varVar[\$targetOffset - \$hostOffset]"
+    local varVar="${pkgAccumVarVars[$(( hostOffset + 1 ))]}"
```

Comment:
BTW this can be `${vars[hostOffset + 1]}`, because whatever is within `${a[i]}` is parsed like arithemetic.  Maybe that is too subtle for some but I'm used to it.

(Also now I notice the naming style is sometimes `camelCase` and sometimes `role_host`)

I guess it is pretty confusing because below you there is some "lazy evaluation" with `${varVar}[  $(( expr )) ]`, which indeed does need `$(( ))`, because it's constructing a string to be evaluated later.  This code is very confusing :)

Reply:
I don't like these references that much, but it seems that logic is in various places, I'm still trying to find a "better" solution, but I have to be able to understand the code holistically and I'm not there.